invenio_app_ils/circulation/search.py

```diff
@@ -88,6 +89,9 @@ def _default_parser(qstr=None):
     query_string = request.values.get('q', '')
     query = _default_parser(qstr=query_string)
 
+    if not current_user.is_authenticated:
```

Comment:
Just a minor comment isn't it nicer to start with this check and raise as soon as possible?

Reply:
@topless can you also change in line 90 to something like
```
parser = query_parser or _default_parser
query = parser(qstr=query_string)
```
as we ignore I think the the `query_parser` argument.

Reply:
@zzacharo I can but @equadon can too :)

Reply:
@zzacharo It would be nicer but then we can't raise a `UnauthorizedSearchError` since it expects the query string. Should we raise another exception instead?